dart/lib/src/sentry.dart

```diff
@@ -20,6 +21,7 @@ typedef AppRunner = FutureOr<void> Function();
 /// Sentry SDK main entry point
 class Sentry {
   static Hub _hub = NoOpHub();
+  static const PlatformChecker _platformChecker = PlatformChecker();
```

Comment:
I'd add this to the `SentryOptions`, as we use the options as a service locator, for example:

https://github.com/getsentry/sentry-dart/blob/main/dart/lib/src/sentry_options.dart#L44-L50

the idea would be to do something similar, so we don't need to pass the `PlatformChecker` around nor haing a static instance.

Reply:
Moved the platform checker to options:  https://github.com/getsentry/sentry-dart/blob/b91b0c68f853505822e0b1e14a729fd100dde463/dart/lib/src/sentry_options.dart#L284

Inject it as an optional parameter in the init function, as we need to set it on options before calling `_initDefaultValues `: https://github.com/getsentry/sentry-dart/blob/b91b0c68f853505822e0b1e14a729fd100dde463/dart/lib/src/sentry.dart#L38

Is this ok, or do you have maybe have a better suggestion? 🤔

Reply:
answered here https://github.com/getsentry/sentry-dart/pull/202#discussion_r532870839